src/server/app.js

```diff
@@ -89,7 +89,12 @@ else {
 }
 app.use(express.static(path.join(rootDir, 'static')));
 
-/* Set up sessions, using Redis in production and default in-memory for testing environment*/
+// Set up Redis, which is used for sessions and to cache requests to external APIs
+// eslint-disable-next-line no-console
+redisClient.connect().catch(redisError => { console.error('Redis error:', redisError); });
```

Comment:
Previously, the `connect` call was in the block below, making sure we don't need redis when running tests.
If I'm not mistaken, now it will try to connect in all cases, which might just mean we get an extra error when running tests.
Probably not a big deal in itself, but since we're not expecting to have an instance of redis in that case, it would probably be better to keep the connection in the `if (process.env.NODE_ENV !== 'test')` block

Reply:
Agreed, I had been thinking that it might be causing issues if the cache is not always available, but I will test how the Wikipedia helpers behave when `redisClient` is not connected and adapt their code if necessary.
And I think I had also mixed up 'test' and 'development' when things were not working as expected yesterday and I wanted to exclude one potential issue...

Reply:
By the way, is there any specific reason to use `console.log` instead of `debug` [here](https://github.com/metabrainz/bookbrainz-site/commit/e949ae64c4cc90385eb6592ef95dfb4f9e815e77#diff-c9e676a021d3cb1d0e5e7ae832276cbfebab683b325852042a9b0aa9b9e8d11cR72)?

Reply:
> if the cache is not always available

*A* cache should always be available. When we don't set `sessionOptions.store`, the cache will be in memory instead (which is why we should remove the redis connection).

> any specific reason to use console.log

None at all, that's just unclean code from me !
I assume I copied the example connection string from the library examples and didn't correct it to use `debug`

Reply:
> A cache should always be available

I was more worried about the API cache not being available than the session cache, but I've wrapped the Redis commands with a condition in 6538335e9f7af3fe1dd0b5c4665572ae68be04f4.

> I assume I copied the example connection string from the library examples and didn't correct it to use debug

Logging has now been standardized in 5d873671eb54030e7071bde465b73507bf5eba2a, after I've squashed my previous Redis refactoring commits into 239977633d3d9d331f3d617e0fbec89033ec39b1. I've also reused the same Redis setup code for the BB API app and extracted the whole duplicated session setup code into a new module in 6c4f3b96a67d585940aeebe334ca8a0516327b87.